server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -84,6 +84,7 @@ public class MetadataRolloverService {
         AutoShardingType.COOLDOWN_PREVENTED_DECREASE,
         "es.auto_sharding.cooldown_prevented_decrease.total"
     );
+    private static final String NON_EXISTENT_SOURCE = "_none_";
```

Comment:
Could you elaborate on why you think that (might) make the code cleaner? If we do that, we'd have to change the serialization code and add some checks. Whereas now, we only supply this static value to the constructor in a couple of places and the rest of the code doesn't need to know about it.

Reply:
I thought `null` makes it more explicit that this value might not always exist, instead of having to check if the name of the "index" is `_none_`. Also, having a nullable value already informs the maintainer about this, while now they have to know that this value has a string that represents that there is no previous index.

Also using this name, this is only necessary for serialisation, so it feels to me less leaky to use it only in the code it's needed instead of everywhere. 

What do you think?

Reply:
I get what you mean, and I agree that that would make sense, but we don't actually "check if the name of the index is `_none_`" anywhere right now. So, to me, it feels like we're preparing/making changes for _if_ we ever have to do those checks, which feels a bit premature to me. Are you still in favor of making these changes pre-emptively?

Reply:
That is my preference, but I hear you and I do not feel strongly about it. So you can leave it as is, if you prefer this.